### PR TITLE
WT-6915 Write Architecture Guide for log files.

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

File: src/docs/arch-log-file.dox [hunk 1]
```diff
@@ -1,7 +1,59 @@
...
+All records in the log, including the header, are subject to the minimum record
+size. Typically that is 128 bytes. That value may be changed if direct I/O for
+logging is turned on. When enabling direct I/O via \c direct_io=log to ::wiredtiger_open
+then the minimum alignment is the value specified for \c buffer_alignment.
+In all other cases it is 128 bytes.
+
+The reason that 128 bytes was chosen as the minimum record size was to eliminate any
+possibility of false sharing in memory and invalidating cache lines when multiple
+threads are writing their log records to the memory buffers in parallel. False sharing
+would be a performance impact.
```
Comment (keithbostic) [inline src/docs/arch-log-file.dox:37]: "would be a performance impact" -> "would potentially impact performance"

File: src/docs/arch-log-file.dox [hunk 2]
```diff
@@ -1,7 +1,59 @@
...
+
+From version 2 onward the first record in the log file, typically at offset 128 in the
+file, is a special system log
+record that contains the LSN of the ending of the previous log file. For the first
```
Comment (keithbostic) [inline src/docs/arch-log-file.dox:41]: "of the ending" -> "of the end"

File: src/docs/arch-log-file.dox [hunk 3]
```diff
@@ -1,7 +1,59 @@
...
+log file that value is an invalid LSN. For later log files the LSN indicates the end
+of the previous file so that recovery can detect missing log records and holes in
+log files that appear at the end of a file.
+
+As described in @ref log_subsystem, multiple threads are writing records to the log
+in a lock free manner and multiple log buffer slots may be in flight at any given time.
+A hole can be generated in a log file if a buffer with a later LSN is written before
+a buffer with an earlier LSN. That can happen at a log file boundary too and that is why
```
Comment (keithbostic) [inline src/docs/arch-log-file.dox:49]: "can happen at a log file boundary too" -> "can also happen at a log file boundary"

**Review by keithbostic — APPROVED** [review]

**Comment by sueloverso** [comment]
Thank you @keithbostic. I have fixed all of the suggested wording changes.

**Review by tammybailey — APPROVED** [review]
LGTM

**Pull request merged**